src/SetChunkData.cpp

```diff
@@ -95,14 +94,25 @@ cSetChunkData::cSetChunkData(
 
 void cSetChunkData::CalculateHeightMap(void)
 {
+	// Find the heighest present section in the chunk
+	size_t MaxSection = 0;
+	for (size_t i = cChunkData::NumSections - 1; i != 0; --i)
+	{
+		if (m_ChunkData.GetSection(i) != nullptr)
+		{
+			MaxSection = i;
+			break;
+		}
+	}
+	const int MaxHeight = static_cast<int>(MaxSection + 1) * cChunkData::SectionHeight - 1;
```

Comment:
Couldn't you just use `int` for `MaxSection` and do without the cast?

Reply:
`cChunkData::GetSection` takes a `size_t` so AFAIK there has to be a cast somewhere.

Reply:
Fair enough.